### PR TITLE
RP-3592: Reference doc update for JSONParser plugin

### PR DESCRIPTION
**Pull request by neelesh-nirmal**

### CODE REVIEW

File: transform-plugins/docs/JSONParser-transform.md [hunk 1]
```diff
@@ -6,13 +6,20 @@ Parses an input JSON event into a record. The input JSON event could be either a
 string fields to values or it could be a complex nested JSON structure. The plugin allows you
 to express JSON paths for extracting fields from complex nested input JSON.
 
-## Parsing Simple JSON
+### Use Case
+JSON Parser reads the values of input column record by record, parses it as json and then extracts the values from json as per output schema.
```
Comment (lav-mudgal-guavus) [inline transform-plugins/docs/JSONParser-transform.md:21]: We need to explicitly mention that a values of the input column should be a JSON string.

File: transform-plugins/docs/JSONParser-transform.md [hunk 2]
```diff
@@ -101,12 +110,19 @@ The mappings in the plugin will be:
     | zip               | $.employee.address.zip     |
     | country           | $.employee.address.country |
     +================================================+
+    
+## JSONPath
```
Comment (lav-mudgal-guavus) [inline transform-plugins/docs/JSONParser-transform.md:127]: This should be moved above example section

File: transform-plugins/docs/JSONParser-transform.md [hunk 3]
```diff
@@ -136,14 +152,90 @@ can be only be applied where the expression results in an array.
     | avg      | double | Average value of array of numbers            |
     | stddev   | double | Standard deviation value of array of numbers |
     | length   | int    | Length of the array                          |
-    +==================================================================+ 
+    +==================================================================+
+
+#### Samples example for writing the JSONPath expression
+See [here](https://github.com/json-path/JsonPath#path-examples) for sample expression.
```
Comment (lav-mudgal-guavus) [inline transform-plugins/docs/JSONParser-transform.md]: give full url here. no hyperlink.

File: transform-plugins/docs/JSONParser-transform.md [hunk 4]
```diff
@@ -136,14 +152,90 @@ can be only be applied where the expression results in an array.
...
+
+You can also do a hands on [here](http://jsonpath.herokuapp.com/).
```
Comment (lav-mudgal-guavus) [inline transform-plugins/docs/JSONParser-transform.md]: give full url here. no hyperlink.

**Review by deepak-maheshwari-ext — APPROVED** [review]
LGTM

**Pull request merged**